analysis/src/topObjectSelection.cc

```diff
@@ -128,19 +130,52 @@ vector<TParticle> topObjectSelection::jetSelection() {
     mom.SetPtEtaPhiM(Jet_pt[i], Jet_eta[i], Jet_phi[i], Jet_mass[i]);
     bool hasOverLap = false;
     for (auto lep : recoleps){
-        if (mom.TLorentzVector::DeltaR(lep) < 0.4) hasOverLap = true;
+      if (mom.TLorentzVector::DeltaR(lep) < 0.4) hasOverLap = true;
     }
     if (hasOverLap) continue;
     auto jet = TParticle();
     jet.SetMomentum(mom);
     jet.SetFirstMother(i);
     jets.push_back(jet);
-    for (UInt_t iu = 0; iu < 19; iu++) {
-     // Jet_SF_CSV[iu] *= m_btagSF.getSF(jet, Jet_btagCSVV2[i], Jet_hadronFlavour[i], iu);
+    if (m_isMC) {
+      b_btagCSVV2 = Jet_btagCSVV2[i];
+      BTagEntry::JetFlavor JF = BTagEntry::FLAV_UDSG;
+      if (abs(Jet_hadronFlavour[i]) == 5) {
+        JF = BTagEntry::FLAV_B;
+        double iCSVWgtHF = m_btagSF.eval_auto_bounds("central", JF, abs(Jet_eta[i]), Jet_pt[i], Jet_btagCSVV2[i]);
+        double iCSVWgtHF_up = m_btagSF_up.eval_auto_bounds("up", JF, abs(Jet_eta[i]), Jet_pt[i], Jet_btagCSVV2[i]);
+        double iCSVWgtHF_dn = m_btagSF_dn.eval_auto_bounds("down", JF, abs(Jet_eta[i]), Jet_pt[i], Jet_btagCSVV2[i]);
+        if (iCSVWgtHF != 0) csvWgtHF *= iCSVWgtHF;
+        if (iCSVWgtHF_up != 0) csvWgtHF_up *= iCSVWgtHF_up;
```

Comment:
do iCSVWgtLF == 0 happen?

Reply:
You should maybe check if it happens. It would seem to be an error condition for the scale factor to be 0, so if it does happen we should understand why. If it never happens, then its okay, of course

Reply:
I checked it with using MC samples, It doesn't to be 0.